projects/fsastorefrontlib/src/cms-components/my-account/order/order-history/order-history.component.html

```diff
@@ -1,5 +1,5 @@
 <ng-container *ngIf="orders$ | async as orderHistory">
-  <ng-container *ngIf="orderHistory">
+  <ng-container *ngIf="isLoaded$ | async; else loading">
```

Comment:
Please re-check this with throttling on, because if I'm not mistaken, this will again load that message "You have no orders" first. That's why Spartacus has this `if` in `noOrder` template itself. If that's the case then simply copy their implementation.

Reply:
Checked and it works.